staging/src/k8s.io/cli-runtime/pkg/resource/visitor_test.go

```diff
@@ -197,3 +197,117 @@ func TestFlattenListVisitorWithVisitorError(t *testing.T) {
+			},
+		},
+		{
+			name:           "Non-recursive with default file extensions",
+			root:           "../../artifacts/visit-order",
+			recursive:      false,
+			fileExtensions: FileExtensions,
+			expectedVisitorPaths: []string{
+				"../../artifacts/visit-order/02.yaml",
+				"../../artifacts/visit-order/10.yaml",
+				"../../artifacts/visit-order/2.yaml",
+				"../../artifacts/visit-order/a.yaml",
+				"../../artifacts/visit-order/b.yml",
+			},
+		},
+		{
+			name:           "Recursive with yaml file extension",
+			root:           "../../artifacts/visit-order",
+			recursive:      true,
+			fileExtensions: []string{".yaml"},
+			expectedVisitorPaths: []string{
+				"../../artifacts/visit-order/0/10.yaml",
+				"../../artifacts/visit-order/0/a/10.yaml",
+				"../../artifacts/visit-order/02.yaml",
+				"../../artifacts/visit-order/10.yaml",
+				"../../artifacts/visit-order/2.yaml",
+				"../../artifacts/visit-order/a/a.yaml",
+				"../../artifacts/visit-order/a.yaml",
+			},
+		},
+		{
+			name:           "Recursive with json and yml file extensions",
+			root:           "../../artifacts/visit-order",
+			recursive:      true,
+			fileExtensions: []string{".json", ".yml"},
+			expectedVisitorPaths: []string{
+				"../../artifacts/visit-order/a/b.json",
+				"../../artifacts/visit-order/b.yml",
+			},
+		},
+		{
+			name:           "Non-recursive with json and yml file extensions",
+			root:           "../../artifacts/visit-order",
+			recursive:      false,
+			fileExtensions: []string{".json", ".yml"},
+			expectedVisitorPaths: []string{
+				"../../artifacts/visit-order/b.yml",
+			},
+		},
+		{
+			name:                 "Non-existent file extensions should return nothing",
+			root:                 "../../artifacts/visit-order",
+			recursive:            true,
+			fileExtensions:       []string{".foo"},
+			expectedVisitorPaths: []string{},
+		},
+		{
+			name:                 "Non-existent root should return file not found error",
+			root:                 "../../artifacts/does/not/exist",
+			recursive:            true,
+			fileExtensions:       []string{".foo"},
+			expectedVisitorPaths: []string{},
+			expectedError:        "lstat ../../artifacts/does/not/exist: no such file or directory",
+		},
+		{
+			name:           "Visitor for single file is returned even if extension does not match",
+			root:           "../../artifacts/visit-order/a.yaml",
+			recursive:      true,
+			fileExtensions: []string{"foo"},
+			expectedVisitorPaths: []string{
+				"../../artifacts/visit-order/a.yaml",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			visitors, err := ExpandPathsToFileVisitors(nil, tt.root, tt.recursive, tt.fileExtensions, nil)
+			if err != nil {
+				assert.Equal(t, tt.expectedError, err.Error())
+				return
+			}
+			assert.Equal(t, len(tt.expectedVisitorPaths), len(visitors))
+			for i, v := range visitors {
+				assert.Equal(t, tt.expectedVisitorPaths[i], v.(*FileVisitor).Path)
+			}
```

Comment:
this will avoid panicking if expectedVisitorPaths is shorter than visitors:

```go
actualPaths := []string{}
for _, v := range visitors {
  actualPaths = append(actualPaths, v.(*FileVisitor).Path)
}
assert.Equal(t, tt.expectedVisitorPaths, actualPaths)
```

Reply:
Good idea. It prints a better assert error this way. I made the change.